builtin/mainmenu/tab_online.lua

```diff
@@ -285,14 +316,46 @@ local function search_server_list(input)
 	-- Search the serverlist
 	local search_result = {}
 	for i, server in ipairs(serverlistmgr.servers) do
-		local name_matches, description_matches = true, true
-		for _, keyword in ipairs(keywords) do
+		local name_matches, description_matches, filter_matches = true, true, true
```

Comment:
At this point we might want to extract this entire thing into a local function `matches_query(server, query)`. The `filter_matches = false; break`s below could just be `return false`s then.

Reply:
Done, now it's in a `matches_query` function.